brightener.cpp

```diff
@@ -1,16 +1,20 @@
 #include "brightener.h"
 
-ImageBrightener::ImageBrightener(std::unique_ptr<Image> inputImage) : m_inputImage(std::move(inputImage)) {
+ImageBrightener::ImageBrightener(std::shared_ptr<Image> inputImage) : m_inputImage(inputImage) {
 }
 
-bool ImageBrightener::ValidateImage()
+/// <summary>
+/// This is newly added method which encapsulate
+///  validateBoundry & Brigtining whole image.
+/// </summary>
+/// <returns>attenuatedPixelCount</returns>
+int ImageBrightener::BrightenWholeImageWithInRange()
 {
-	if (m_inputImage->m_columns <= 1024 && m_inputImage->m_rows <= 1024) {
-		return true;
-	}
-	else {
-		return false;
+	if (m_inputImage->ValidateBoundry())
+	{
+		return BrightenWholeImage();
 	}
+	return -1;
```

Comment:
-1 expresses error here. But a consumer of this class would know that only if they looked at the implementation.
If they miss this detail, they still get an integer and continue - resulting in hard-to-investigate errors elsewhere in the flow.
Why not throw an exception?